test/test_boundary_conditions_integration.jl

```diff
@@ -13,7 +13,7 @@ function test_boundary_condition(arch, FT, topo, side, field_name, boundary_cond
                                 buoyancy=SeawaterBuoyancy(), tracers=(:T, :S))
 
     success = try
-        time_step!(model, 1e-16, euler=true)
```

Comment:
why `euler=true` is gone? or why was it there?

Reply:
AB2 has memory (one reason why I don't like it). If you don't know the status of the tendencies (ie if you cannot guarantee that they were computed consistently at the previous time step), then you need to write `euler=true` to use `time_step!(model)`. For most users running a simulation, this is not an issue because `run!` handles it. In typical circumstances, we can tell when somebody is taking the first time step (we check the previous time step size now; previously we used `clock.iteration`).

But this PR changes the default to RK3. RK3 doesn't have an `euler` option for `time_step`.